tests/generative/test_query_model.py

```diff
@@ -0,0 +1,357 @@
+import os
+
+import hypothesis as hyp
+import hypothesis.errors
+import hypothesis.strategies as st
+import sqlalchemy.orm
+
+from databuilder.query_model import (
+    AggregateByPatient,
+    Filter,
+    Function,
+    PickOneRowPerPatient,
+    Position,
+    SelectColumn,
+    SelectPatientTable,
+    SelectTable,
+    Sort,
+    TableSchema,
+    ValidationError,
+    Value,
+    has_many_rows_per_patient,
+    has_one_row_per_patient,
+)
+
+from ..conftest import QueryEngineFixture
+from ..lib.in_memory import InMemoryDatabase, InMemoryQueryEngine
+from .conftest import count_nodes, observe_inputs
+
+next_id = iter(range(1, 2**63)).__next__
+
+
+Base = sqlalchemy.orm.declarative_base()
+
+
+class P1(Base):
+    __tablename__ = "p1"
+    Id = sqlalchemy.Column(sqlalchemy.Integer, primary_key=True, default=next_id)
+    PatientId = sqlalchemy.Column(sqlalchemy.Integer)
+    i = sqlalchemy.Column(sqlalchemy.Integer)
+    b = sqlalchemy.Column(sqlalchemy.Boolean)
```

Comment:
By not having two columns of the same type in any table, are we missing out on some of the space of possible queries?

Reply:
Yes. I simplified it as much as I could to start with and then didn't add them back. It would be straightforward to add them, but there are lots of ways we could make it richer. I dunno. WDYT?

Reply:
I've made this change since it was so easy.